language/diem-framework/modules/AccountAdministrationScripts.move

```diff
@@ -593,5 +594,27 @@ module AccountAdministrationScripts {
             Errors::INVALID_ARGUMENT,
             Errors::ALREADY_PUBLISHED;
     }
+
+    /// # Summary
+    /// Publishes a `DiemIdDomains` resource under a VASP account.
+    /// The sending account must be a VASP account, and be a parent VASP account.
+    ///
+    /// # Technical Description
+    /// Publishes a `DiemId::DiemIdDomains` resource under `account`. It then
+    /// The `domains` field is a vector of DiemIdDomain, and will be empty on initialization.
+    ///
+    /// # Parameters
+    /// | Name      | Type     | Description                                           |
+    /// | ------    | ------   | -------------                                         |
+    /// | `account` | `signer` | The signer of the sending account of the transaction. |
+    ///
+    /// # Common Abort Conditions
+    /// | Error Category              | Error Reason                      | Description                                                                                   |
+    /// | ----------------            | --------------                    | -------------                                                                                 |
+    /// | `Errors::ALREADY_PUBLISHED` | `DiemId::EDIEMIDDOMAIN`           | A `DiemId::DiemIdDomains` resource has already been published under `account`.     |
+
+    public(script) fun create_diem_id_domains(account: signer) {
```

Comment:
Yeah need for one-off calls on VASPs that already exist

Reply:
Is there not a way to try and create it if it doesn't exist when adding a domain?

Reply:
The hard part here is in `update_diem_id_domains` since that only takes a TC signer currently. However, in order to publish under the parent VASP's account we also need the parent VASP's signer due to the memory model in Move.

There are a couple ways I can see of us being able to remove this script:
* AOS will need to send admin scripts for each parent VASP that is already created to upgrade those accounts
* `update_diem_id_domains` could take both the TC signer and the parent VASP signer, however that means this transaction would then need to be a multi-agent transaction (support is landing for this in Diem very soon, but I'm not sure about when the off-chain infrastructure would be able to support this?)

IMO the simplest thing to do here if we want to get rid of this script is the former of these, so I'd probably recommend that if we wanted to remove this script.